readthedocs/doc_builder/environments.py

```diff
@@ -224,9 +224,12 @@ def run(self):
             self.exit_code = cmd_ret['ExitCode']
 
             # Docker will exit with a special exit code to signify the command
-            # was killed due to memory usage, make the error code nicer.
-            if (self.exit_code == DOCKER_OOM_EXIT_CODE and
-                    self.output == 'Killed\n'):
+            # was killed due to memory usage, make the error code
+            # nicer. Sometimes the kernel kills the command and Docker doesn't
+            # not use the specific exit code, so we check if the word `Killed`
+            # is in the last 15 lines of the command's output
+            killed_in_output = 'Killed' in '\n'.join(self.output.splitlines()[-15:])
+            if self.exit_code == DOCKER_OOM_EXIT_CODE or killed_in_output:
                 self.output = _('Command killed due to excessive memory '
```

Comment:
I don't like to override the output of the command here, I think that it should be good to have a specific exception for this and show the error similar than other errors (like timeout, for example)

Reply:
I believe 'Killed' is normally the last line of output. Was this not the case for your build? We might be able to severely reduce the number of false positives here by limiting this.

Reply:
Also, do we know if this string changes if you run Sphinx with another locale?

Reply:
Also also, perhaps logic should be:

```python
if self.exit_code == DOCKER_OOM_EXIT_CODE or self.exit_code == -1 and killed_in_output
```

I'm not sure if -1 is correct here

Reply:
> I believe 'Killed' is normally the last line of output. Was this not the case for your build?

No.

My last lines of the output are:

```
    building 'lxml.etree' extension
    creating build/temp.linux-x86_64-2.7
    creating build/temp.linux-x86_64-2.7/src
    creating build/temp.linux-x86_64-2.7/src/lxml
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/include/libxml2 -I/tmp/pip-build-_MvJYS/lxml/src/lxml/includes -I/usr/include/python2.7 -c src/lxml/lxml.etree.c -o build/temp.linux-x86_64-2.7/src/lxml/lxml.etree.o -w
    x86_64-linux-gnu-gcc: internal compiler error: Killed (program cc1)
    Please submit a full bug report,
    with preprocessed source if appropriate.
    See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 4
    
    ----------------------------------------
Command "/home/humitos/rtfd/code/readthedocs.org/user_builds/rtd/envs/latest/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-_MvJYS/lxml/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-3RRzJB-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/humitos/rtfd/code/readthedocs.org/user_builds/rtd/envs/latest/include/site/python2.7/lxml" failed with error code 1 in /tmp/pip-build-_MvJYS/lxml/
```

>  Also, do we know if this string changes if you run Sphinx with another locale?

This is a good question. But I think the locale would be the one used by pip, since it's a command ran by pip but at the end it's `x86_64-linux-gnu-gcc` what's ran and if the LANG is not exported I would say that it's the system's default one used.

> I'm not sure if -1 is correct here

Nop. My `exit_code` was `1` for this particular case.

Reply:
Ah yes, pip rather. Not sure if there is a good way to get errors out of pip.

We should check for exit_code == 1 in this logic

Reply:
I can add that.

Do you think that it's better to append our error message to the `self.output` instead of replacing it?

So, in my case, instead of just `Command killed due to excessive memory consumption` it would have some information where the user can see that it gets killed because of `lxml` and maybe it's something that's not needed to build the docs, so the user can mock it.